itools-packager/src/main/java/com/powsybl/itools/ItoolsPackagerMojo.java

```diff
@@ -79,33 +81,125 @@ public void setFiles(File[] files) {
+        Objects.requireNonNull(baseDir);
+        Objects.requireNonNull(fileName);
+
+        String extension = compareFileExtension(fileName);
+        String newFileName = fileName;
+        // invalid extension
+        if (extension.equals("")) {
+            extension = ".zip";
+            newFileName = packageName + extension;
+        }
+
+        Path packageFilePath = baseDir.resolve(newFileName);
+        FileOutputStream fos = new FileOutputStream(String.valueOf(packageFilePath));
+
+        getLog().info("Generate package file : " + newFileName);
+
+        switch (extension) {
+            case ".zip":
+                try (ZipArchiveOutputStream zos = new ZipArchiveOutputStream(fos)) {
+                    Files.walkFileTree(dir, new SimpleFileVisitor<Path>() {
+                        @Override
+                        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                            ZipArchiveEntry entry = new ZipArchiveEntry(baseDir.relativize(file).toString());
+                            if (Files.isExecutable(file)) {
+                                entry.setUnixMode(0100770);
+                            } else {
+                                entry.setUnixMode(0100660);
+                            }
+                            zos.putArchiveEntry(entry);
+                            Files.copy(file, zos);
+                            zos.closeArchiveEntry();
+                            return FileVisitResult.CONTINUE;
+                        }
+
+                        @Override
+                        public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+                            zos.putArchiveEntry(new ZipArchiveEntry(baseDir.relativize(dir).toString() + "/"));
+                            zos.closeArchiveEntry();
+                            return FileVisitResult.CONTINUE;
+                        }
+                    });
+                }
+                break;
+
+            case ".tar.gz":
+                try (TarArchiveOutputStream tos = new TarArchiveOutputStream(new GZIPOutputStream(new BufferedOutputStream(fos)))) {
+                    tos.setBigNumberMode(TarArchiveOutputStream.BIGNUMBER_STAR); // to store big number in the archive.
+                    tos.setLongFileMode(TarArchiveOutputStream.LONGFILE_GNU); // to store long file names in the archive.
+
+                    Files.walkFileTree(dir, new SimpleFileVisitor<Path>() {
+                        @Override
+                        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                            TarArchiveEntry entry = new TarArchiveEntry(new File(file + "/"));
+                            entry.setName(baseDir.relativize(file).toString());
+                            entry.setSize(file.toFile().length());
+                            entry.setMode(0770);
```

Comment:
You have to check if the file is executable or not, to choose between 0770 and 0660 (see. above)

Reply:
The tests performed on Linux and Windows show that compressed binary files are already executable files. 
0660 is not part of this package ?
I added this choice between 0770 and 0660.